lib/BrickMapController.js

```diff
@@ -942,12 +942,12 @@ function BrickMapController(options) {
                     }*/
                     //TODO: update the smart contract and after that uncomment the above code and eliminate the following if statement
                     if (!currentAnchoredHashLink) {
-                        anchoring.getAllVersions(keySSI, (err, versions) => {
+                        anchoring.getLastVersion(keySSI, (err, version) => {
                             if (err) {
                                 return OpenDSUSafeCallback(listener)(createOpenDSUErrorWrapper(`Failed to retrieve versions of anchor`, err));
                             }
 
-                            if (versions && versions.length === 0) {
+                            if (!version) {
```

Comment:
my comment is not related to the current fix... it is about the logic of this if block statement, that i just noticed: we have a currentAnchoredHashLink but we can't confirm it with the getLastVersion call. How and why?

Reply:
I can't say because i haven't implemented this section, I've just adapted the method call